### PR TITLE
Add doc for HighFive::DataSpace

### PR DESCRIPTION
**Pull request by alkino**

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
       with:
         submodules: 'recursive'
 
-    - uses: mamba-org/provision-with-micromamba@main
+    - uses: mamba-org/setup-micromamba@v1
```
Comment (1uc) [inline .github/workflows/gh-pages.yml:27]: This has already been merged, you need to rebase.

Comment (alkino) [reply .github/workflows/gh-pages.yml:27]: I don't think so: https://github.com/BlueBrain/HighFive/blob/master/.github/workflows/gh-pages.yml#L27

Comment (alkino) [reply .github/workflows/gh-pages.yml:27]: Only ci.yml has been modified this is gh-pages.yml (for the doc)

Comment (1uc) [reply .github/workflows/gh-pages.yml:27]: In that case: Thank you! =)

**Review by 1uc — APPROVED** [review]

**Pull request merged**